GoRogue/MapGeneration/Region.cs

```diff
@@ -178,302 +132,67 @@ public Region(Point northWest, Point northEast, Point southEast, Point southWest
         /// Returns a string detailing the region's corner locations.
         /// </summary>
         public override string ToString()
-            => $"Region: NW{NorthWestCorner}=> NE{NorthEastCorner}=> SE{SouthEastCorner}=> SW{SouthWestCorner}";
+        {
+            var answer = "Region: ";
+            foreach (var corner in Area.Corners)
+                answer += $"{corner} => ";
+            return answer;
+        }
 
         /// <summary>
         /// Is this Point one of the corners of the Region?
         /// </summary>
         /// <param name="position">the point to evaluate</param>
-        public bool IsCorner(Point position)
-            => position == NorthEastCorner || position == NorthWestCorner || position == SouthEastCorner || position == SouthWestCorner;
+        public bool IsCorner(Point position) => Area.Corners.Contains(position);
```

Comment:
This seems like it might be a more useful function if it was placed directly on `PolygonArea` and forwarded here?